src/test/test_bitcoin.cpp

```diff
@@ -105,6 +105,7 @@ TestingSetup::~TestingSetup()
 
 TestChain100Setup::TestChain100Setup() : TestingSetup(CBaseChainParams::REGTEST)
 {
+    UpdateVersionBitsParameters(Consensus::DEPLOYMENT_SEGWIT, 0, 999999999999ULL);
```

Comment:
For anyone curious, the problem with these tests is that they create a block via ```CreateAndProcessBlock()```, which uses ```BlockAssembler::CreateNewBlock()```, then it adds transactions to the new block. The ```IncrementExtraNonce``` corrects for the new merkle root, but not the new witness root. ```GenerateCoinbaseCommitment``` wouldn't work as it only adds a commitment if there's not one there already.

Updating the params here is fine imo, but if anyone would prefer to avoid it, we'd just need a ```ForceUpdateWitnessCommitment()``` function.

Reply:
Added a comment.